src/test/java/com/sap/sgs/phosphor/fosstars/model/score/AverageCompositeScoreTest.java

```diff
@@ -0,0 +1,176 @@
+    final double secondValue = 2.0;
+
+    final double firstConfidence = 7.3;
+    final double secondConfidence = 6.1;
+
+    ScoreValue firstPreCalculatedScoreValue = new ScoreValue(new FirstScore());
+    firstPreCalculatedScoreValue.set(firstValue);
+    firstPreCalculatedScoreValue.confidence(firstConfidence);
+
+    ScoreValue secondPreCalculatedScoreValue = new ScoreValue(new SecondScore());
+    secondPreCalculatedScoreValue.set(secondValue);
+    secondPreCalculatedScoreValue.confidence(secondConfidence);
+
+    ScoreValue scoreValue = score.calculate(
+        firstPreCalculatedScoreValue,
+        secondPreCalculatedScoreValue);
+
+    assertNotNull(scoreValue);
+
+    double expectedScore = (firstValue + secondValue) / 2;
+    double expectedConfidence = (firstConfidence + secondConfidence) / 2;
+
+    assertEquals(expectedScore, scoreValue.get(), PRECISION);
+    assertEquals(expectedConfidence, scoreValue.confidence(), PRECISION);
+  }
+
+  @Test
+  public void oneScorePreCalculated() {
+    Score score = new TestAverageCompositeScore();
+
+    final double firstValue = 3.0;
+    final double firstConfidence = 7.2;
+
+    ScoreValue preCalculatedScoreValue = new ScoreValue(new FirstScore());
+    preCalculatedScoreValue.set(firstValue);
+    preCalculatedScoreValue.confidence(firstConfidence);
+
+    ScoreValue scoreValue = score.calculate(preCalculatedScoreValue);
+    assertNotNull(scoreValue);
+
+    double expectedScore = (firstValue + SecondScore.VALUE) / 2;
+    assertEquals(expectedScore, scoreValue.get(), PRECISION);
+
+    double expectedConfidence = (firstConfidence + Confidence.MAX) / 2;
+    assertEquals(expectedConfidence, scoreValue.confidence(), PRECISION);
+  }
+
+  private static class FirstScore extends FeatureBasedScore {
+
+    private static final Feature<Double> FEATURE = new DoubleFeature("first feature");
+
+    private static final double VALUE = 0.2;
+
+    FirstScore() {
+      super("First score", FEATURE);
+    }
+
+    @Override
+    public ScoreValue calculate(Value... values) {
+      return scoreValue(VALUE, values);
+    }
+
+    @Override
+    public int hashCode() {
+      return getClass().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj.getClass() == getClass();
+    }
+  }
+
+  private static class SecondScore extends FeatureBasedScore {
+
+    private static final Feature<Double> FEATURE = new DoubleFeature("second feature");
+
+    private static final double VALUE = 0.5;
+
+    SecondScore() {
+      super("Second score", FEATURE);
+    }
+
+    @Override
+    public ScoreValue calculate(Value... values) {
+      return scoreValue(VALUE, values);
+    }
+
+    @Override
+    public int hashCode() {
+      return getClass().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj.getClass() == getClass();
+    }
+  }
+
+  private static class TestAverageCompositeScore extends AverageCompositeScore {
+
```

Comment:
Please remove this line.

Reply:
I would prefer to keep this like since it separates the class definition and fields.

Reply:
ok